### PR TITLE
Adc unregister

### LINKED ISSUE #4683: Sensor: unregister on the fly
have CLT assigned to any ADC input
set input to NONE, click 'burn'
go to lua

```
cltSensor = Sensor.new("clt")
```

expected behaviour: happy script 

actual result:
```
2022-10-19_19_52_40_965: EngineState: LUA ERROR loading script: [string "-- start of technical boring details please i..."]:50: Tried to create a Lua sensor of type Clt, but one was already registered.

```

at the moment full power cycle is needed for real analog sensor to un-register

### PR DESCRIPTION
**Pull request by rusefillc**
Sensor: unregister on the fly fix #4683

### CODE REVIEW

File: firmware/hw_layer/adc/adc_subscription.cpp [hunk 1]
```diff
@@ -95,6 +95,10 @@ static AdcSubscriptionEntry* findEntry() {
 	efiSetPadUnused(getAdcChannelBrainPin("adc unsubscribe", entry->Channel));
 #endif // EFI_PROD_CODE
 
+	if (entry->Sensor) {
```
Comment (mck1117) [inline firmware/hw_layer/adc/adc_subscription.cpp:98]: we don't need to use `entry->Sensor`, we can just use the `sensor` parameter (which should is `== entry->Sensor`)

Comment (mck1117) [reply firmware/hw_layer/adc/adc_subscription.cpp:98]: And then also don't need a branch.

File: firmware/hw_layer/adc/adc_subscription.cpp [hunk 2]
```diff
@@ -95,6 +95,8 @@ static AdcSubscriptionEntry* findEntry() {
 	efiSetPadUnused(getAdcChannelBrainPin("adc unsubscribe", entry->Channel));
 #endif // EFI_PROD_CODE
 
+	entry->Sensor->unregister();
```
Comment (mck1117) [inline firmware/hw_layer/adc/adc_subscription.cpp:98]: no no, just do `sensor.unregister();`

Comment (rusefillc) [reply firmware/hw_layer/adc/adc_subscription.cpp]: Sorry, Netflix and past midnight on east coast

Comment (mck1117) [reply firmware/hw_layer/adc/adc_subscription.cpp]: > Netflix and past midnight

it happens!

**Pull request merged**